app/src/sharedTest/java/org/oppia/android/app/topic/revision/TopicRevisionFragmentTest.kt

```diff
@@ -286,6 +292,40 @@ class TopicRevisionFragmentTest {
     }
   }
 
+  @Test
+  fun testTopicRevisionFragment_fragmentLoaded_verifyCorrectArgumentsPassed() {
+    launchTopicActivityIntent(
+      internalProfileId = internalProfileId,
+      classroomId = TEST_CLASSROOM_ID_1,
+      topicId = FRACTIONS_TOPIC_ID
+    ).use { scenario ->
+      clickRevisionTab()
+      testCoroutineDispatchers.runCurrent()
+      scenario.onActivity { activity ->
+
+        val topicFragment = activity.supportFragmentManager
+          .findFragmentById(R.id.topic_fragment_placeholder) as TopicFragment
+        val viewPager = topicFragment.requireView()
+          .findViewById<ViewPager2>(R.id.topic_tabs_viewpager)
+        val topicRevisionFragment = topicFragment.childFragmentManager
+          .findFragmentByTag("f${viewPager.currentItem}") as TopicRevisionFragment
+
+        val receivedInternalProfileId = topicRevisionFragment
+          .arguments?.extractCurrentUserProfileId()?.internalId ?: -1
+        val args = topicRevisionFragment.arguments?.getProto(
+          TopicRevisionFragment.TOPIC_REVISION_FRAGMENT_ARGUMENTS_KEY,
+          TopicRevisionFragmentArguments.getDefaultInstance()
+        )
+        val receivedTopicId = checkNotNull(args?.topicId) {
+          "Expected topic ID to be included in arguments for TopicRevisionFragment."
+        }
```

Comment:
FYI in most of the new test code you can remove the extra check*() logic and just have Truth do the work. E.g. in this case:

```kotlin
assertThat(args?.topicId).isEqualTo(FRACTIONS_TOPIC_ID)
```

Since Truth is a Java API, it can take null values (and will fail correctly if it receives one since, in this case, ``FRACTIONS_TOPIC_ID`` is not null and will trigger a failure if ``args`` is null).